test/integration/test_migrations.py

```diff
@@ -150,3 +150,25 @@ def test_compatibility_modified(self):
         assert t.load_home(compatibility_path) == "Modified file"
         # not Removed because compatibility was modified
         assert "POTATO" in t.load_home(cppstd_compat_path)
+
+    def test_compatibility_modified_by_conan(self):
+        t = TestClient(light=True)
+        t.run("-v")
+        compatibility_path = "extensions/plugins/compatibility/compatibility.py"
+        old = t.load_home(compatibility_path)
+        # In old versions of Conan, assume POTATO was present in the file
+        new = old + "\n# This file was generated by Conan POTATO"
+        t.save_home({"version.txt": "2.11",
+                     compatibility_path: new})
+        # But now with the newer version, Conan removed it
+        # so the triggered migration should remove it
+        t.run("-v")
```

Comment:
Does this show a migration message? Maybe check that it is printed?
```suggestion
        t.run("-v")
        assert "Successfully updated compatibility.py" in t.out
```